lib/ansible/modules/cloud/amazon/aws_glacier_vault.py

```diff
@@ -0,0 +1,239 @@
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: aws_glacier_vault
+short_description: Manage Glacier vaults in AWS
+description:
+    - Manage Glacier vaults in AWS
+version_added: "2.6"
+requirements: [ boto3 ]
+author: "Loic Blot <loic.blot@unix-experience.fr>"
+options:
+  name:
+    description:
+      - Name of the glacier vault
+    required: true
+    default: null
+  state:
+    description:
+      - Create or remove the glacier vault
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  tags:
+    description:
+      - A hash/dictionary of tags to add to the new instance or to add/remove from an existing one.
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+RETURN = '''
+vault:
+    description: the vault object
+    returned: always
+    type: complex
+    contains:
+        creation_date:
+            description: The vault creation date.
+            returned: always
+            type: string
+            sample: "2018-02-13T22:10:28.590Z"
+        number_of_archives:
+            description: Number of archives currently in the vault.
+            returned: always
+            type: int
+            sample: 0
+        size_in_bytes:
+            description: Current vault size.
+            returned: always
+            type: int
+            sample: 0
+        vault_arn:
+            description: The vault AWS ARN.
+            returned: always
+            type: string
+            sample: "arn:aws:glacier:eu-west-3:123457788994:vaults/test-vault"
+        vault_name:
+            description: The vault name.
+            returned: always
+            type: string
+            sample: "test-vault"
+tags:
+    description: the vault tags
+    returned: always
+    type: dictionary
+    sample: {
+        "testtag": "testvalue"
+    }
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create a simple glacier vault
+- aws_glacier_vault:
+    name: myglaciervault
+
+# Create a simple glacier vault in eu-west-3 region
+- aws_glacier_vault:
+    name: myglaciervault
+    region: eu-west-3
+
+# Remove an glacier vault
+- aws_glacier_vault:
+    name: myglaciervault
+    state: absent
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (ec2_argument_spec, boto3_conn, HAS_BOTO3, get_aws_connection_info,
+                                      compare_aws_tags, camel_dict_to_snake_dict)
+
+try:
+    import botocore
+    from botocore.exceptions import ClientError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+class GlacierVaultManager(object):
+    def __init__(self, connection, module):
+        self.connection = connection
+        self.module = module
+        self.name = self.module.params.get('name')
+
+    def list(self):
+        vaults = {}
+        try:
+            vaults = self.connection.list_vaults()
```

Comment:
A paginator exists for list_vaults - you should use it: http://boto3.readthedocs.io/en/latest/reference/services/glacier.html#Glacier.Paginator.ListVaults

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#api-throttling-and-pagination